### PR TITLE
Change serialization of `CqlDuration` to ISO-8601 duration compliant String

### LINKED ISSUE #1617: Output `duration` values as ISO-8601, not Cassandra "standard" notation
Currently API Tables handling of CQL `duration` type:

1. Accepts ISO-8601 duration formats (`PnYnMnDTnHnMnS`, like "PT1H30M", see https://en.wikipedia.org/wiki/ISO_8601) -- as well as Cassandra's "Standard" format (like "1h30m")
2. Produces Cassandra style values.

But specification defines that output should be done using ISO-8601 format.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Changes JSON serialization of CQL duration type (internally `CqlDuration`) to use ISO-8601 Period notation, instead of Cassandra native notation.

**Which issue(s) this PR fixes**:
Fixes #1617

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    // Check if we have time part
+    if (nanoSeconds == 0L) {
+      if (sb.length() == 1) {
+        return "PT0S"; // All zeroes case
+      }
+      return sb.toString(); // Only date part exists
+    }
+
+    sb.append('T');
+    long hours = nanoSeconds / NANOS_PER_HOUR;
```
Comment (vkarpov15) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java:51]: Neat trick, looks like dividing 2 longs gives you the value rounded down :+1:

Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java:51]: Yes, in that's the logic in Java (and other C-derivative languages) wrt integer division.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    // Seconds more challenging due to possibility of fractional seconds
+    if (nanoSeconds > 0L) {
+      long seconds = nanoSeconds / NANOS_PER_SECOND;
+      sb.append(seconds);
+      final long fractionalNanos = nanoSeconds - (seconds * NANOS_PER_SECOND);
+      if (fractionalNanos > 0L) {
+        String nanoString = String.format("%09d", fractionalNanos);
+        if (nanoString.endsWith("0")) {
+          nanoString = nanoString.replaceAll("0+$", "");
```
Comment (Yuqi-Du) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java:71]: This is cool!

File: src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        // Identity
+        Arguments.of("P1Y2M3DT4H5M6S", "P1Y2M3DT4H5M6S"),
+
+        // Actual conversions; manual examples:
+        Arguments.of("1h30m", "PT1H30M"),
+        Arguments.of("3h7s", "PT3H7S"),
+        Arguments.of("22h1ms", "PT22H0.001S"),
+        Arguments.of("1m37us", "PT1M0.000037S"),
+
+        // Actual conversions; inspired by ChatGPT:
```
Comment (Yuqi-Du) [inline src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverterTest.java:42]: This is one of the moments that we are glad of having AI. 😊

**Review by Yuqi-Du — APPROVED** [review]
LGTM!

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  private static final long NANOS_PER_HOUR = 60 * NANOS_PER_MINUTE;
+
+  /**
+   * Convert {@link CqlDuration} to ISO-8601 duration string ("PnYnMnDTnHnMnS").
+   *
+   * @param value CqlDuration to convert to ISO-8601 duration string
+   * @return ISO-8601 compliant duration string
+   */
+  public static String toISO8601Duration(CqlDuration value) {
+    int months = value.getMonths();
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java:21]: Yeah not sure: I think they store these as two 64-bit `long`s (one for Date part, other Time), but then why separate Months and Days? (there probably is a good reason... maybe months having variable number of days but not sure)

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java (hunk 2 above)
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java:71]: Confession to make: this was written by ChatGPT!

File: src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverterTest.java (hunk 3 above)
Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverterTest.java:42]: Indeed. I did have to fix couple of small things, but overall it helped.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (hemidactylus) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/CqlDurationConverter.java:21]: Yes, this is the reason: months / days / sub-days are three incommensurable quantities
(months have 28-31 days, days have 23-25 hours). Durations are a conceptual quantity, not translatable 1:1 to a time span (which is also why they are not-sortable).

**Pull request merged**